### ISSUE TITLE
[Feature request] Add support for plugins?

### ISSUE THREAD

**Issue by nexfwall** [flatpak/flatpak#90]
As we all know, some applications can have third-party or even builtin plugins, which extend their functionality or allow applications to have modular structure. I can install a plugin for "Open in Terminal" button in my Nautilus, installing gnome-terminal-nautilus rpm package in Fedora. But can I do the same thing using flatpak, without hacking the container on my own? (Which I believe is the same bad thing, like doing `make install` without packaging it in rpm/dpkg.)

Some frameworks/libraries have plugins too, to extend their functionality. And some of them are third-party. As a example - Qt.

Can flatpak handle this somehow?

Maybe some kind of separate installation type (.plugin, for example), which can be linked/binded to the application/framework container before starting it up. With possibility to selectively disable one of _plugins_ without deleting and possibility to install them from the repository. Maybe this _type_ can even have it's own dependencies.

Or there are other ways to achieve that purpose?

_// I'm sorry if my English is not fine enough._

**Comment by matthiasclasen** [comment]
This is what extensions are meant to provide. We haven't fully fleshed out extensions for applications; but there are some examples already: The .Locale extension that both applications and runtimes use is an example.

**Comment by nexfwall** [comment]
>This is what extensions are meant to provide. We haven't fully fleshed out extensions for applications; but there are some examples already: The .Locale extension that both applications and runtimes use is an example.

If _extensions_ provide everything I had pointed to (can be disabled, have it's own dependencies), then, maybe we should close this issue as a duplicate of _extensions_?

**Comment by alexlarsson** [comment]
Yeah, this feature request essentially describes extensions. We might need the tooling to improve, but the core feature is there. So, i'll close this for now.
